src/dbnode/persist/fs/retriever.go

```diff
@@ -580,6 +687,16 @@ func (reqs *shardRetrieveRequests) resetQueued() {
 	reqs.queued = reqs.queued[:0]
 }
 
+type emptyIndexChecksum struct{}
+
+var _ block.StreamedChecksum = (*emptyIndexChecksum)(nil)
+
+func (*emptyIndexChecksum) RetrieveIndexChecksum() (ident.IndexChecksum, error) {
+	return ident.IndexChecksum{}, nil
+}
+
+var emptyChecksum = &emptyIndexChecksum{}
```

Comment:
Better to just make `func (*emptyIndexChecksum) RetrieveIndexChecksum() (ident.IndexChecksum, error)` be `func (emptyIndexChecksum) RetrieveIndexChecksum() (ident.IndexChecksum, error)` then you don't need to make it a pointer type for this var?